src/esm.ts

```diff
@@ -48,21 +49,35 @@ function _genStatement (type: 'import' | 'export' | 'import type' | 'export type
 
   const namesStr = _names.map(i => i.as ? `${i.name} as ${i.as}` : i.name).join(', ')
   if (nameArray) {
-    return `${type} { ${namesStr} } from ${genString(specifier, opts)};`
+    return `${type} { ${namesStr} } from ${genString(specifier, opts)}${_getAssertClause(type, opts.assert)};`
   }
-  return `${type} ${namesStr} from ${genString(specifier, opts)};`
+  return `${type} ${namesStr} from ${genString(specifier, opts)}${_getAssertClause(type, opts.assert)};`
+}
+
+function _getAssertClause (type: ImportExportType, assert?: { type: string }) {
+  // import assertions isn't specified type-only import or export on Typescript
+  if (type === 'import type' || type === 'export type') {
+    return ''
+  }
+  return assert ? ` assert { type: ${genString(assert.type)} }` : ''
```

Comment:
We need to check for `assert?.type`

Reply:
I've fixed it!
https://github.com/unjs/knitwork/pull/34/commits/a8307b45127145ab527079d15755ad0af093eb72